src/components/LHNOptionsList/OptionRowLHN.js

```diff
@@ -57,8 +57,11 @@ const OptionRowLHN = (props) => {
     const optionItem = SidebarUtils.getOptionData(props.reportID);
 
     React.useEffect(() => {
+        if (!optionItem || !optionItem.isPinned) {
+            return;
+        }
```

Comment:
This will not hide context menu when it's open with pinned state and then unpinned on another device